### DISCUSSION TITLE
Guide to complete change the containerd root path

### DISCUSSION THREAD

**Discussion by busyboy77** [k3s-io/k3s#6675] [Q&A]
Can someone please guide me how I can use the `config.toml.tmpl` to completely confine or change the containerd deployment directory.

the github link of https://github.com/k3s-io/k3s/blob/master/pkg/agent/templates/templates_linux.go doesn't make any sense to me to do or change the containerd deployment other than /var/lib/containerd.

regards

**Comment by brandond** [comment]
What specifically are you trying to change? The containerd root and state dirs are not passed via the templated config, but cli flags:
https://github.com/k3s-io/k3s/blob/454440f9a3b585bd038293f9de5acb470380c3fa/pkg/agent/containerd/config_linux.go#L35-L39

The values for these are set at:
* https://github.com/k3s-io/k3s/blob/454440f9a3b585bd038293f9de5acb470380c3fa/pkg/agent/config/config_linux.go#L13
* https://github.com/k3s-io/k3s/blob/454440f9a3b585bd038293f9de5acb470380c3fa/pkg/agent/config/config.go#L465

**Reply by asloan7** [reply]
@brandond so you can do something like this?
`curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC=" --state=/newpath/containerd-state --data-dir=/newpath/rancher/k3s --kubelet-arg root-dir=/newpath/kubelet" sh -`

**Reply by brandond** [reply]
No, you can't change the containerd args. As noted above, they are hardcoded.
